### ISSUE TITLE
Vanity alignment used inside base ctor call

### ISSUE THREAD

**Issue by knocte** [fsprojects/fantomas#2111]
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAXAngB0WAqgHYAWAhofADaLwDqiARgKIAe0i26AlrIWALwAdPhDBcSiAE5d0YVu049Cw4aLCFEAdwAUAZ3Sl0AV11p6zNh268AygeO6ANGMIbJ8q0rQfFvAJQCqmqiwC7EUjJylr6E2jYACtKE6ABiAIxBwVnZOWqCkABCpNAA1ogU6rCy2JIcpLXwYABmsJJg6MRcumDmPtZ8mjLEYHaGJgDaAKRcALr5mbmLi9risvpjun6OC0u7Lm59SgEAvioiEBo64i1oNhGklFwAXoZKAJKELc5wyYgs6Ld0LVSABbcQAcwAwrx0H90AEhOdgqFxOFpLJDrwVp9YN8YXCTpBHOAoLpEOhuIRwbpUGBxiSIKAspAACRYXC0yCrIkM0SsgBuDyMyDQ9JyAAZiTlIB8kMkbM9kFLsgAWXkQGa847KxnqqBsnAiqDcnV8lmCyjC2li7JpU1qSAAWVILAAMuJEK7yuCOjycmkAEzivWatTa3lM4Ks9lGyDleAAeSa7o0dkw1D9UfNQqNNqyAftfKYFCTKaVeogkEoTUgIa1psjDoNHLQkAYsFglEzTYtVtFFbAAGZC5WPmTJKlxA8AHJaR4abvBIHCuthht66OGzntzuLyvZy25gcqkdQO5guCUXgAQXQxcTyY9e9ETQeZNXonDakbZpj247XaFgKObWgOACsp6QDY2DFIgBSIC0tTxPUoLklIz4QMuiAfhAX4hBuzaxjugEEb2R45AAbJB0GwfBiGerAmhSNApBkh8-KwCx-QYe0kgrlkoafuuzKEf%2Bu5AQefZ0gOADs1EwewdGtIg%2BDYLgkgsWxhAcVxSg8a%2BlDvgJ9YRgRf6tsRz7AYeoE5AAHPJtEIcpkKUKxujQoQ%2Bh8dA6CtPpb7YcZa6mSJ5lQJZElkbZ2QAJyOYpzm1I6iAggw6GngZRnBIJuHCVm4VtgBVmSeRtqSgOUEKXBSWINCV6EAFhlBTlJnfmZW4WcVUUgf2-p2pVNHsNeTSwpI0IgiCpA8VhOFgHhuphZ1EXdaRvXSf6BaDdVSm1OeXCXrwTXZWouXzflTaFZFa02X1trDttsEjWN%2B2HY1p6zcFQmhQVy1FeJN1SXmwRpCelWyuU6AJoQAAqkiYLQQzHS1p1tfhS0titAMidFd1ZGkEGPcNkiwEYFAACKII8YJjTNfEo6IZ0LWAP77oVJqA2V%2BNUZVzosG8TQw%2BEhBMM1NjEK06CI-AvqniqwZfXlP2XX9HM4%2BtwNqGkcm8y6HxNFwLAJuphitKwNSILouh6aeYEK61IXtRjsZq1muMbbaDm6ywABKiBwJIdBcDLxA8fLc3M6z%2Brs8kJXu5rohpPF3t%2BwH8DTkYaVSEmbywiCNKnmkEcXb%2Bf0gkYlDcPOiApK0U0UlIMNbj1t0e-mFXSqnrTwI6FdVx6teSPXtOQdAZCSMUY0APqDCHtaK%2Bdyul5jXKxy3QMDgGA3Snz16SBPmAJpI7r6NLsvHvbqOO%2Bjv0r67PYa5vW07y6e8H0fJ-oBnWeSDnecFwOIuC9I4dRXuXSuXBq6D2Ho3ZunMYr5getKN%2BpBD7Hy6OgXuECoF10MCPSqY8UK%2BSkDPYOvpi5LzZqrNe8C8bBADGDF%2BLAABqOYCjiHgBCM%2BodTx2UvozNGi1b4uxoerVuCcIABkJkwlIZNfJKHYRQLhZCeEXwoU7YRnJ75mnjpvHmTDyZVAAOLknNrUK2ShuE8Ttuom%2BKsV7XTERvHIAYdY7z7pAj0BQrylAKBPUo5JdDQxsKheqmd3oDiygzDUgiWagKIqtJxXN6Fe2lLOTQ1d4LoCYuUJuuBKYG0IDIJQ14KApR-gAnIUTbFCPsQk7GbtH4uOTtKAA0ogDgAshblA%2BCEkEnonyZUCjUuJzstGiMaeIgcg4O7ZCdC6UWYJdDECsXLfhMTr61OXvUkiSSEHBEHNvOZCoqTUD3uCTOkNzCOgQOWKpwzgElyoQ4xJkznHZEHM-OZ15HjgkILIwg8jbBcD%2BWMWoMNYAQ2SK8I6QzmojKjpuF5DSH5TJyIOJB3zK5SEIK8fkDEqTlPSpIAp4him8EqdkapjzKHRz%2Bo4t5yS1CDkYXMrB3AyyulBAweApBXKwGtlSdJ1dkYIviWJXZjL9nMukXM8mXRSAMGoIsroxAbCYGhSwUVNKNF1IlXHJpHz9FzPaZ0igkMPh%2BPIGPbVDtvq6u2fq9eTLRCDjcXM7x5AShll0HvUmFBZz6BoOyzxGgzGW2thSum-E7VKwdc8nZBq0UfNSR6%2Bou1EDyt0NAaQYJcWwngEQJQlNoBuQntxOFJ0BGbNGZorqKKdGGqyIOFpxygQHUwbc21V9PwkhmMqLkugUjW1pNU44QA)

#### Code

```fsharp
type UnhandledWebException =
    inherit Exception

    new(status: WebExceptionStatus, innerException: Exception) =
        { inherit Exception(SPrintF1
                                "Backend not prepared for this WebException with Status[%i]"
                                (int status),
                            innerException) }

    new(info: SerializationInfo, context: StreamingContext) =
        { inherit Exception(info, context) }
```
            

#### Result

```fsharp
type UnhandledWebException =
    inherit Exception

    new(status: WebExceptionStatus, innerException: Exception) =
        { inherit Exception(SPrintF1 "Backend not prepared for this WebException with Status[%i]" (int status),
                            innerException) }

    new(info: SerializationInfo, context: StreamingContext) = { inherit Exception(info, context) }

```
            
#### Problem description

Similar issue as https://github.com/fsprojects/fantomas/issues/1442 .

#### Expected result?

Maybe this (not sure):

```fsharp
type UnhandledWebException =
    inherit Exception

    new(status: WebExceptionStatus, innerException: Exception) =
        { inherit
            Exception(
                SPrintF1 "Backend not prepared for this WebException with Status[%i]" (int status),
                innerException
            ) }

    new(info: SerializationInfo, context: StreamingContext) = { inherit Exception(info, context) }
```

#### Extra information

- [ ] The formatted result breaks by code.
- [ ] The formatted result gives compiler warnings.
- [x] I or my company would be willing to help fix this.

#### Options

Fantomas master branch at 1/1/1990

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

**Comment by nojaf** [comment]
Target 4.7 branch to fix.

**Comment by nojaf** [comment]
> I'm not sure if it aligns with the style guide.

Well, it is not mentioned in the style guide.
The only thing relevant seems:

> Long record field expressions should use a new line and have one indent from the opening {:

To remove all doubts it should be brought up in https://github.com/fsharp/fslang-design#style-guide

**Comment by knocte** [comment]
Actually I think this is a very similar case to this fantomas conversion:
```
{ Foo = Bar;SomeMemberOfRecord = SomeFuncCall("1111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111", otherParam)}
```
->
```
{ Foo = Bar
  SomeMemberOfRecord =
    SomeFuncCall(
        "1111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111",
        otherParam
    ) }
```

So the analog expected result would be:

```
type UnhandledWebException =
    inherit Exception

    new(status: WebExceptionStatus, innerException: Exception) =
        { inherit
            Exception(
                SPrintF1 "Backend not prepared for this WebException with Status[%i]" (int status),
                innerException
            ) }

    new(info: SerializationInfo, context: StreamingContext) = { inherit Exception(info, context) }
```

**Comment by knocte** [comment]
> Target 4.7 branch to fix.

Hey @nojaf, given that 4.7 has been merged, what branch to target now, master?

**Comment by nojaf** [comment]
I'll create a `4.8` branch. Could you please also bring it up in the style?

**Comment by nojaf** [comment]
This is available in the 5.1 series.
